src/pages/Recipe/Recipe.tsx

```diff
@@ -0,0 +1,165 @@
+import styles from './Recipe.module.scss';
+import Text from 'components/Text';
+import NaviIcon from 'components/icons/NaviIcon'
+import InfoGrid from './components/InfoGrid';
+import NeedList from './components/NeedList';
+
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import qs from 'qs';
+import { useParams, Link } from "react-router-dom";
+import { createContext, Dispatch, SetStateAction, useContext } from 'react';
+import parse from 'html-react-parser';
+import { Images } from 'App/App';
+
+export interface Ingradients {
+    id: number,
+    name: string,
+    amount: number,
+    unit: string
+}
+
+export interface Equipments {
+    id: number,
+    name: string
+}
+
+interface Directions {
+    id: number,
+    description: string
+}
+
+export interface RecipeInfo {
+    name: string,
+    preparationTime?: number,
+    cookingTime?: number,
+    totalTime?: number,
+    likes?: number,
+    servings?: number,
+    rating?: number,
+    summary: string,
+    ingradients: Ingradients[],
+    equipments: Equipments[],
+    directions: Directions[]
+}
+
+export interface RecipeInfoContextType {
+    recipeCont: RecipeInfo;
+    setRecipeInfo: Dispatch<SetStateAction<RecipeInfo>>;
+}
+
+export const RecipeInfoContext = createContext<RecipeInfoContextType>({
+    recipeCont: {
+        name: '',
+        summary: '',
+        ingradients: [],
+        equipments: [],
+        directions: []
+    },
+    setRecipeInfo: () => { }
+});
+
+export const useRecipeInfoContext = () => useContext(RecipeInfoContext);
+
+
+function Recipe() {
+
+    const { documentId } = useParams();
+
+    const [recipe, setRecipe] = useState<RecipeInfo>(
+        {
+            name: '',
+            summary: '',
+            ingradients: [],
+            equipments: [],
+            directions: []
+        });
+
+    const [imgRecipe, setImgRecipe] = useState<Images>();
+
+    useEffect(() => {
+        const fetch = async () => {
+            try {
+                const params = {
+                    populate: ['ingradients', 'equipments', 'directions.image', 'images', 'category']
```

Comment:
давай вынесем этот массив в константу над компонентом, например const POPULATE_ITEMS:
```suggestion
                    populate: POPULATE_ITEMS
```